components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -1812,125 +1816,341 @@ private int handleStartIndexEqualsNULL(Integer startIndex) {
+            if (StringUtils.isEmpty(domainName)) {
+                domainName = applyFilterPropertiesForDomain(node);
+            }
+            return domainName;
+        } catch (BadRequestException e) {
+            String errorMessage = String
+                    .format(" Domain name in the attribute value: %s does not match with the domain parameter: %s in "
+                            + "the request.", node.getValue(), domainName);
+            throw new CharonException(errorMessage, e);
+        }
+    }
+
+    /**
+     * Method to check isFilterUsersAndGroupsOnlyFromPrimaryDomainEnabled() or isFilteringEnhancementsEnabled() which
+     * enables filtering in primary domain only.
+     *
+     * @param node Expression node.
+     * @return Primary domain name if properties are enabled or return NULL when properties are disabled.
+     */
+    private String applyFilterPropertiesForDomain(ExpressionNode node) {
+
+        if (isFilterUsersAndGroupsOnlyFromPrimaryDomainEnabled()) {
+            return UserCoreConstants.PRIMARY_DEFAULT_DOMAIN_NAME;
+        } else if (isFilteringEnhancementsEnabled()) {
+
+            // To maintain backward compatibility.
+            if (SCIMCommonConstants.EQ.equalsIgnoreCase(node.getOperation())) {
+                if (StringUtils.equals(node.getAttributeValue(), SCIMConstants.GroupSchemaConstants.DISPLAY_NAME_URI)) {
+                    return UserCoreConstants.PRIMARY_DEFAULT_DOMAIN_NAME;
+                }
+            }
+        }
+        // Domain value should be returned to indicate no special requirement for primary user store filtering.
+        return null;
+    }
+
+    /**
+     * Method to resolve domain name if a domain is attached to the attribute value.
+     *
+     * @param domainName Domain name in the request.
+     * @param node       Expression Node.
+     * @return Domain name
+     */
+    private String resolveDomainInAttributeValue(String domainName, ExpressionNode node) throws BadRequestException {
+
+        String attributeName = node.getAttributeValue();
+        String attributeValue = node.getValue();
+        String extractedDomain;
+
+        // Checks whether the attribute name is display name, display uri or value uri.
+        if (StringUtils.equals(attributeName, SCIMConstants.GroupSchemaConstants.DISPLAY_NAME_URI) || StringUtils
+                .equals(attributeName, SCIMConstants.GroupSchemaConstants.DISPLAY_URI) || StringUtils
+                .equals(attributeName, SCIMConstants.GroupSchemaConstants.VALUE_URI)) {
+
+            // Identify whether there is a domain separator in the attribute value.
+            int indexOfDomainSeparator = attributeValue.indexOf(CarbonConstants.DOMAIN_SEPARATOR);
+
+            // Index less than 1 would indicate that there is not domain appended in front if the attribute value.
+            if (indexOfDomainSeparator > 0) {
+                if (log.isDebugEnabled())
```

Comment:
Wrap the debug log with curly brackets

Reply:
Fixed: 0811c8c408e0d35596e71d82f167868b3f079ae7